exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -114,11 +114,17 @@ func metricDataToSplunk(logger *zap.Logger, data pdata.Metrics, config *Config)
 	return splunkMetrics, numDroppedTimeSeries, nil
 }
 
-func timestampToEpochMilliseconds(ts *timestamppb.Timestamp) float64 {
+func timestampToEpochMilliseconds(ts *timestamppb.Timestamp) *float64 {
 	if ts == nil {
-		return 0
+		return nil
 	}
-	return float64(ts.GetSeconds()) + math.Round(float64(ts.GetNanos())/1e6)/1e3
+
+	val := float64(ts.GetSeconds()) + math.Round(float64(ts.GetNanos())/1e6)/1e3
+	if val == 0 {
+		return nil
+	}
```

Comment:
Add a comment to explain what this does.  I am not sure this is needed. What case does this handle?

Reply:
I've added a comment here. If I'm not mistaken, it's to handle the case when the event has a timestamp, but it's 0 (as opposed to having no timestamp at all)